From: Arturo Buzarra <arturo.buzarra@digi.com>
Date: Wed, 11 Dec 2024 17:20:22 +0100
Subject: [PATCH] config_board: add support to STM32MP255 processor

Signed-off-by: Arturo Buzarra <arturo.buzarra@digi.com>
---
 .../files/resources-files/config_board.sh     | 32 +++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/resources-files/config_board.sh b/resources-files/config_board.sh
index a9ce24b..a37981a 100644
--- a/resources-files/config_board.sh
+++ b/resources-files/config_board.sh
@@ -13,6 +13,7 @@ STM32MP2="stm32mp2"
 STM32MP135="stm32mp135"
 STM32MP157="stm32mp157"
 STM32MP157FEV1="stm32mp157f-ev1st"
+STM32MP255="stm32mp255"
 STM32MP257="stm32mp257"
 STM32MP257FEV1="stm32mp257f-ev1st"
 
@@ -79,6 +80,37 @@ if [[ "$COMPATIBLE" == *"$STM32MP157"* ]]; then
   fi
 fi
 
+if [[ "$COMPATIBLE" == *"$STM32MP255"* ]]; then
+  SEMANTIC_SEGMENTATION_MODEL="deeplabv3/deeplabv3_257_int8_per_tensor$NN_EXT"
+  SEMANTIC_SEGMENTATION_LABEL="deeplabv3/labels_pascalvoc"
+  SEMANTIC_SEGMENTATION_DATA="deeplabv3/testdata/"
+  POSE_ESTIMATION_DATA="yolov8n_pose/testdata/"
+  POSE_ESTIMATION_MODEL="yolov8n_pose/yolov8n_256_quant_pt_uf_pose_coco-st$NN_EXT"
+  FACE_DETECTION_MODEL="blazeface/blazeface_128x128_quant$NN_EXT"
+  FACE_DETECTION_DATA="blazeface/testdata/"
+  FACE_RECO_MODEL="facenet/facenet512_160x160_quant$NN_EXT"
+  FACE_RECO_DATA="facenet/testdata/"
+  FACE_DATABASE="database/"
+  # STM32MP255 settings
+  MACHINE=$STM32MP255
+  DWIDTH=640
+  DHEIGHT=480
+  DFPS=30
+  COMPUTE_ENGINE="--npu"
+  IMAGE_CLASSIFICATION_MODEL="mobilenet/mobilenet_v2_1.0_224_int8_per_tensor$NN_EXT"
+  IMAGE_CLASSIFICATION_LABEL="mobilenet/labels_imagenet_2012"
+  IMAGE_CLASSIF_DATA="mobilenet/testdata/"
+  if [[ "$NN_EXT" == ".nb" ]]; then
+    OBJ_DETEC_MODEL="coco_ssd_mobilenet/ssd_mobilenet_v2_fpnlite_10_256_int8_per_tensor$NN_EXT"
+    OBJ_DETEC_MODEL_LABEL="coco_ssd_mobilenet/labels_coco_dataset_80"
+  else
+    OBJ_DETEC_MODEL="coco_ssd_mobilenet/ssd_mobilenet_v2_fpnlite_10_256_int8$NN_EXT"
+    OBJ_DETEC_MODEL_LABEL="coco_ssd_mobilenet/labels_coco_dataset_80"
+  fi
+  OBJ_DETECT_DATA="coco_ssd_mobilenet/testdata/"
+  OPTIONS="--dual_camera_pipeline"
+fi
+
 if [[ "$COMPATIBLE" == *"$STM32MP257"* ]]; then
   SEMANTIC_SEGMENTATION_MODEL="deeplabv3/deeplabv3_257_int8_per_tensor$NN_EXT"
   SEMANTIC_SEGMENTATION_LABEL="deeplabv3/labels_pascalvoc"
-- 
2.47.1

