From: Isaac Hermida <isaac.hermida@digi.com>
Date: Thu, 21 Apr 2016 14:32:57 +0200
Subject: [PATCH] osdep_adf.h: fix for undefined ath_sysctl_pktlog_size

Signed-off-by: Isaac Hermida <isaac.hermida@digi.com>
---
 CORE/SERVICES/COMMON/osdep_adf.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CORE/SERVICES/COMMON/osdep_adf.h b/CORE/SERVICES/COMMON/osdep_adf.h
index 708b6f89fa03..48ca78618840 100644
--- a/CORE/SERVICES/COMMON/osdep_adf.h
+++ b/CORE/SERVICES/COMMON/osdep_adf.h
@@ -168,6 +168,10 @@ typedef unsigned long TQUEUE_ARG;
 #define ATH_SYSCTL_PROC_DOSTRING(ctl, write, filp, buffer, lenp, ppos) \
     proc_dostring(ctl, write, filp, buffer, lenp, ppos)

+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,17,0)
+typedef struct ctl_table ctl_table;
+#endif
+
 /*
  * Byte Order stuff
  */
