From: Isaac Hermida <isaac.hermida@digi.com>
Date: Mon, 27 Jun 2016 13:07:23 +0200
Subject: [PATCH] Kbuild: do not create an auxiliar p2p on init

Signed-off-by: Isaac Hermida <isaac.hermida@digi.com>
---
 Kbuild | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Kbuild b/Kbuild
index 339262874c94..b14417a558cd 100644
--- a/Kbuild
+++ b/Kbuild
@@ -1254,7 +1254,7 @@ CDEFINES += -DQCA_HT_2040_COEX
 else

 #Open P2P device interface only for non-Mobile router use cases
-CDEFINES += -DWLAN_OPEN_P2P_INTERFACE
+# CDEFINES += -DWLAN_OPEN_P2P_INTERFACE

 #Enable 2.4 GHz social channels in 5 GHz only mode for p2p usage
 CDEFINES += -DWLAN_ENABLE_SOCIAL_CHANNELS_5G_ONLY
