From: Isaac Hermida <isaac.hermida@digi.com>
Date: Mon, 17 Oct 2016 10:06:18 +0200
Subject: [PATCH] Add .gitignore rules

Signed-off-by: Isaac Hermida <isaac.hermida@digi.com>
---
 .gitignore | 9 +++++++++
 1 file changed, 9 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 000000000000..9886b5284c3f
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,9 @@
+*.o
+*.o.cmd
+.tmp_versions
+.*
+Module.symvers
+modules.order
+*.mod.c
+cscope.*
+wlan.ko
