From: Alex Gonzalez <alex.gonzalez@digi.com>
Date: Thu, 13 Dec 2012 11:34:01 +0100
Subject: [PATCH] Make crypto optional.

Signed-off-by: Alex Gonzalez <alex.gonzalez@digi.com>
---
 Makefile |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b8bc7d3..208ebbd 100644
--- a/Makefile
+++ b/Makefile
@@ -35,7 +35,8 @@ LDLIBS += `pkg-config --libs openssl`
 
 reglib.o: keys-ssl.c
 
-else
+endif
+ifeq ($(USE_GCRYPT),1)
 CFLAGS += -DUSE_GCRYPT
 LDLIBS += -lgcrypt
 
