From: Javier Viguera <javier.viguera@digi.com>
Date: Wed, 20 Apr 2011 11:13:10 +0200
Subject: [PATCH] del-mdev_regulatory

Signed-off-by: Javier Viguera <javier.viguera@digi.com>
---
 util-linux/mdev.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/util-linux/mdev.c b/util-linux/mdev.c
index c4829a5..c7b038d 100644
--- a/util-linux/mdev.c
+++ b/util-linux/mdev.c
@@ -94,6 +94,7 @@
 //usage:       "to match $SEQNUM variable. This prevents plug/unplug races.\n"
 //usage:       "To activate this feature, create empty /dev/mdev.seq at boot."
 
+#include <syslog.h>
 #include "libbb.h"
 #include "xregex.h"
 
@@ -826,6 +827,17 @@ int mdev_main(int argc UNUSED_PARAM, char **argv)
 					load_firmware(fw, temp);
 			}
 		}
+		/* DigiEL: Handle regulatory domain change events */
+		else if (strcmp(action, "change") == 0) {
+			char *modalias;
+			modalias = getenv("MODALIAS");
+			if (strcmp(modalias, "platform:regulatory") == 0) {
+				char *crda = xasprintf("%s", "/sbin/crda");
+				system(crda);
+				free(crda);
+				syslog(LOG_INFO, "[mdev] change regulatory domain [%s]", getenv("COUNTRY"));
+			}
+		}
 
 		if (seq) {
 			xopen_xwrite_close("mdev.seq", utoa(xatou(seq) + 1));
