From: Javier Viguera <javier.viguera@digi.com>
Date: Thu, 31 Jan 2013 11:23:37 +0100
Subject: [PATCH] enable-libnl3

Signed-off-by: Javier Viguera <javier.viguera@digi.com>
---
 Makefile        | 8 +++++++-
 nl80211_utils.c | 4 ++--
 nl80211_utils.h | 2 +-
 3 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 115ad33..3170f65 100644
--- a/Makefile
+++ b/Makefile
@@ -56,17 +56,23 @@ LIBS=		-ldbus-1 -lpthread -lrt -lbluetooth
 
 NL1FOUND := $(shell $(PKG_CONFIG) --atleast-version=1 libnl-1 && echo Y)
 NL2FOUND := $(shell $(PKG_CONFIG) --atleast-version=2 libnl-2.0 && echo Y)
+NL3FOUND := $(shell $(PKG_CONFIG) --atleast-version=3 libnl-3.0 && echo Y)
 
 ifeq ($(NL1FOUND),Y)
+CFLAGS += -DCONFIG_LIBNL1
 NLLIBNAME = libnl-1
 endif
 
 ifeq ($(NL2FOUND),Y)
-CFLAGS += -DCONFIG_LIBNL20
 LIBS += -lnl-genl
 NLLIBNAME = libnl-2.0
 endif
 
+ifeq ($(NL3FOUND),Y)
+LIBS += -lnl-genl
+NLLIBNAME = libnl-3.0
+endif
+
 #ifeq ($(NLLIBNAME),)
 #$(error Cannot find development files for any supported version of libnl)
 #endif
diff --git a/nl80211_utils.c b/nl80211_utils.c
index 4689044..2957f2e 100644
--- a/nl80211_utils.c
+++ b/nl80211_utils.c
@@ -9,7 +9,7 @@
 #include <net/if.h>
 #include "nl80211_utils.h"
 
-#ifndef CONFIG_LIBNL20
+#ifdef CONFIG_LIBNL1
 /* libnl 2.0 compatibility code */
 
 static inline int __genl_ctrl_alloc_cache(struct nl_sock *h, struct nl_cache **cache)
@@ -24,7 +24,7 @@ static inline int __genl_ctrl_alloc_cache(struct nl_sock *h, struct nl_cache **c
 #define nl_socket_alloc nl_handle_alloc
 #define nl_socket_free nl_handle_destroy
 
-#endif /* CONFIG_LIBNL20 */
+#endif /* CONFIG_LIBNL1 */
 
 int iw_debug = 0;
 
diff --git a/nl80211_utils.h b/nl80211_utils.h
index 6a0dbd7..6bbeb2c 100644
--- a/nl80211_utils.h
+++ b/nl80211_utils.h
@@ -11,7 +11,7 @@
 #include <netlink/genl/family.h>
 #include <netlink/genl/ctrl.h>
 
-#ifndef CONFIG_LIBNL20
+#ifdef CONFIG_LIBNL1
 #define nl_sock nl_handle
 #endif
 
