From: Javier Viguera <javier.viguera@digi.com>
Date: Mon, 20 Mar 2017 20:14:09 +0100
Subject: [PATCH] CMakeLists: add option to build without ICU support

Signed-off-by: Javier Viguera <javier.viguera@digi.com>

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 47d54d6..9717a82 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -191,6 +191,8 @@ endif()
 # libicu is highly recommended for RSCALE support
 #  libicu can be found at http://www.icu-project.org
 #  RSCALE info at https://tools.ietf.org/html/rfc7529
+option(WITH_LIBICU "Build with ICU support. Libicu is highly recommended for RSCALE support")
+if(WITH_LIBICU)
 find_package(ICU)
 set_package_properties(ICU PROPERTIES
   TYPE RECOMMENDED
@@ -213,6 +215,7 @@ endif()
 if(ICU_I18N_FOUND)
   set(HAVE_LIBICU_I18N 1)
 endif()
+endif()

 # compile in Berkeley DB support
 find_package(BDB)
