From: Isaac Hermida <isaac.hermida@digi.com>
Date: Tue, 17 Dec 2024 18:34:48 +0100
Subject: [PATCH] check vela return code

Signed-off-by: Isaac Hermida <isaac.hermida@digi.com>
---
 download_models.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/download_models.py b/download_models.py
index 462f7b755eea..fc5ab234ae47 100644
--- a/download_models.py
+++ b/download_models.py
@@ -103,7 +103,11 @@ def convert_model(model_dir, vela_dir):
         if name.endswith(".tflite"):
             print('Converting', name)
             model = os.path.join(model_dir, name)
-            os.system('vela ' + model + " --output-dir " + vela_dir)
+            ret = os.system('vela ' + model + " --output-dir " + vela_dir)
+            if ret != 0:
+                print(f"Error: Failed to convert model {name} with vela")
+                import sys
+                sys.exit(1) 
 
 model_dir = 'models'
 vela_dir = 'vela_models'
