From: Abdur Rehman <abdur_rehman@mentor.com>
Date: Wed, 26 Aug 2015 19:18:11 +0500
Subject: [PATCH 2/3] tcf-agent: obey LDFLAGS

Signed-off-by: Abdur Rehman <abdur_rehman@mentor.com>
---
 agent/Makefile.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/agent/Makefile.inc b/agent/Makefile.inc
index cdf8536dbf9b..70f4710dcacd 100644
--- a/agent/Makefile.inc
+++ b/agent/Makefile.inc
@@ -114,7 +114,7 @@ NO_LINK_F	?= -c
 # Linker definition and flags
 
 LINK		?= $(CC)
-LINK_FLAGS	?= $(CFLAGS)
+LINK_FLAGS	?= $(LDFLAGS) $(CFLAGS)
 LINK_OUT_F	?= $(OUT_OBJ_F)
 
 # Archiver definition and flags
