From: Javier Viguera <javier.viguera@digi.com>
Date: Tue, 16 Apr 2013 18:46:31 +0200
Subject: [PATCH] cross-compile

Signed-off-by: Javier Viguera <javier.viguera@digi.com>
---
 Makefile | 43 +++++++++++++++++++++++++++----------------
 1 file changed, 27 insertions(+), 16 deletions(-)

diff --git a/Makefile b/Makefile
index 3170f65..22137d7 100644
--- a/Makefile
+++ b/Makefile
@@ -25,13 +25,13 @@ SBINDIR ?= $(PREFIX)/sbin
 MANDIR ?= $(PREFIX)/share/man
 
 PKG_CONFIG ?= pkg-config
-ifdef V210
-CC=		$(ATH_CROSS_COMPILE_TYPE)gcc
-LD=		$(ATH_CROSS_COMPILE_TYPE)ld
-else
-CC=		$(ATH_CROSSS_COMPILE_TYPE)gcc
-LD=		$(ATH_CROSSS_COMPILE_TYPE)ld
-endif
+# ifdef V210
+# CC=		$(ATH_CROSS_COMPILE_TYPE)gcc
+# LD=		$(ATH_CROSS_COMPILE_TYPE)ld
+# else
+# CC=		$(ATH_CROSSS_COMPILE_TYPE)gcc
+# LD=		$(ATH_CROSSS_COMPILE_TYPE)ld
+# endif
 SOURCES=abtfilt_main.c \
 		abtfilt_wlan.c \
 		abtfilt_core.c \
@@ -40,19 +40,19 @@ SOURCES=abtfilt_main.c \
 		nl80211_utils.c\
 		btfilter_core.c
 
-INCLUDES=	-Iinclude \
+override INCLUDES += -Iinclude \
 		-Ios/linux/include \
-		-Icommon/include \
-		-I$(V210_DIR)/usr/include/dbus-1.0/ \
-		-I$(V210_DIR)/usr/lib/dbus-1.0/include \
-		-I$(V210_DIR)/usr/local/include/dbus-1.0 \
-		-I$(V210_DIR)/usr/local/lib/dbus-1.0/include \
-		-I$(V210_DIR)/usr/include/bluetooth \
-		-I$(V210_DIR)/usr/local/include/bluetooth \
+		-Icommon/include
+# 		-I$(V210_DIR)/usr/include/dbus-1.0/ \
+# 		-I$(V210_DIR)/usr/lib/dbus-1.0/include \
+# 		-I$(V210_DIR)/usr/local/include/dbus-1.0 \
+# 		-I$(V210_DIR)/usr/local/lib/dbus-1.0/include \
+# 		-I$(V210_DIR)/usr/include/bluetooth \
+# 		-I$(V210_DIR)/usr/local/include/bluetooth \
 
 CFLAGS=		-Wall -g -DABF_DEBUG
 #LIBS=		-ldbus-1 -lpthread -lbtfilt -lrt -lbluetooth
-LIBS=		-ldbus-1 -lpthread -lrt -lbluetooth
+# LIBS=		-ldbus-1 -lpthread -lrt -lbluetooth
 
 NL1FOUND := $(shell $(PKG_CONFIG) --atleast-version=1 libnl-1 && echo Y)
 NL2FOUND := $(shell $(PKG_CONFIG) --atleast-version=2 libnl-2.0 && echo Y)
@@ -80,6 +80,13 @@ endif
 LIBS += $(shell $(PKG_CONFIG) --libs $(NLLIBNAME))
 CFLAGS += $(shell $(PKG_CONFIG) --cflags $(NLLIBNAME))
 
+# dbus
+LIBS += $(shell $(PKG_CONFIG) --libs --static dbus-1)
+CFLAGS += $(shell $(PKG_CONFIG) --cflags dbus-1)
+
+# BlueZ
+LIBS += $(shell $(PKG_CONFIG) --libs bluez)
+CFLAGS += $(shell $(PKG_CONFIG) --cflags bluez)
 
 ifdef BOARD_HAS_ATH_WLAN_AR6004
 CFLAGS += -DMULTI_WLAN_CHAN_SUPPORT
@@ -125,3 +132,7 @@ all: $(OBJECTS)
 
 clean:
 	rm -f $(FILTERAPP) $(OBJECTS)
+
+install:
+	mkdir -p $(DESTDIR)$(SBINDIR)
+	install -m 0755 $(FILTERAPP) $(DESTDIR)$(SBINDIR)
