From: Alex Gonzalez <alex.gonzalez@digi.com>
Date: Mon, 9 Apr 2012 18:20:01 +0200
Subject: [PATCH] mfw_isink: Set defaults if no vssconfig found.

Signed-off-by: Alex Gonzalez <alex.gonzalez@digi.com>
Signed-off-by: Javier Viguera <javier.viguera@digi.com>
---
 libs/vss/vss_common.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/libs/vss/vss_common.c b/libs/vss/vss_common.c
index 2044051..a8e8e57 100755
--- a/libs/vss/vss_common.c
+++ b/libs/vss/vss_common.c
@@ -390,6 +390,22 @@ next:
         cs = cs->next;
     }
 
+	if (i == 0) {
+		for (i = 0; i < VD_MAX; i++) {
+			VideoDevice *vd;
+
+			VS_MESSAGE("No vssconfig, setting defaults.\n");
+			vd = &control->devices[i];
+			vd->fbidx = i;
+			vd->main_fbidx = i;
+			vd->fmt = fmt_from_str("RGBP");
+			sprintf(vd->name, "DISP%d", i);
+			vd->vsmax = 4;
+			vd->mode_num = 0;
+			_getVideoDeviceInfo(vd);
+		}
+	}
+
     sconf_free_cs(css);
 
 }
