From 99b66f0a6014d9f9beab8a4d17d39388738d3ce3 Mon Sep 17 00:00:00 2001
From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Thu, 2 Mar 2023 09:18:49 +0100
Subject: [PATCH 44/68] v4l2codecs: add NV12M format to gstv4l2format

Upstream-Status: Pending
---
 sys/v4l2codecs/gstv4l2format.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/v4l2codecs/gstv4l2format.c b/sys/v4l2codecs/gstv4l2format.c
index 604876e..bbfdee2 100644
--- a/sys/v4l2codecs/gstv4l2format.c
+++ b/sys/v4l2codecs/gstv4l2format.c
@@ -33,7 +33,7 @@ struct FormatEntry
 
 static struct FormatEntry format_map[] = {
   {V4L2_PIX_FMT_NV12, 1, GST_VIDEO_FORMAT_NV12, 8, 420},
-  {V4L2_PIX_FMT_NV12M, 1, GST_VIDEO_FORMAT_NV12, 8, 420},
+  {V4L2_PIX_FMT_NV12M, 2, GST_VIDEO_FORMAT_NV12, 8, 420},
   {V4L2_PIX_FMT_YUYV, 1, GST_VIDEO_FORMAT_YUY2, 8, 422},
   {V4L2_PIX_FMT_SUNXI_TILED_NV12, 1, GST_VIDEO_FORMAT_NV12_32L32, 8, 422},
   {V4L2_PIX_FMT_NV12_4L4, 1, GST_VIDEO_FORMAT_NV12_4L4, 8, 420},
-- 
2.25.1

