From: Javier Viguera <javier.viguera@digi.com>
Date: Wed, 20 Apr 2011 11:13:10 +0200
Subject: [PATCH] del-mdev_regulatory

Signed-off-by: Javier Viguera <javier.viguera@digi.com>
---
 util-linux/mdev.c | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/util-linux/mdev.c b/util-linux/mdev.c
index 3d4e135..3293b59 100644
--- a/util-linux/mdev.c
+++ b/util-linux/mdev.c
@@ -96,6 +96,7 @@
 //usage:       "\n"
 //usage:       "If /dev/mdev.log file exists, debug log will be appended to it."
 
+#include <syslog.h>
 #include "libbb.h"
 #include "xregex.h"
 
@@ -249,8 +250,8 @@
 #endif
 
 
-static const char keywords[] ALIGN1 = "add\0remove\0"; // "change\0"
-enum { OP_add, OP_remove };
+static const char keywords[] ALIGN1 = "add\0remove\0change\0";
+enum { OP_add, OP_remove, OP_change };
 
 struct envmatch {
 	struct envmatch *next;
@@ -1131,13 +1132,24 @@ int mdev_main(int argc UNUSED_PARAM, char **argv)
 			if (!fw)
 				make_device(env_devname, temp, op);
 		}
-		else {
+		else if (op == OP_add) {
 			make_device(env_devname, temp, op);
 			if (ENABLE_FEATURE_MDEV_LOAD_FIRMWARE) {
 				if (op == OP_add && fw)
 					load_firmware(fw, temp);
 			}
 		}
+		/* DigiEL: Handle regulatory domain change events */
+		else if (op == OP_change) {
+			char *env_modalias;
+			env_modalias = getenv("MODALIAS");
+			if (strcmp(env_modalias, "platform:regulatory") == 0) {
+				char *crda = xasprintf("%s", "/sbin/crda");
+				system(crda);
+				free(crda);
+				syslog(LOG_INFO, "[mdev] change regulatory domain [%s]", getenv("COUNTRY"));
+			}
+		}
 
 		dbg1("%s exiting", curtime());
 		if (seq_fd >= 0) {
